core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -561,6 +562,20 @@ private void parseCreateTableAsSelect(GenericLexer lexer, CreateTableModel model
         expectTok(lexer, ')');
     }
 
+    private void ensureThatColumnNamesAreUnique(ObjList<QueryColumn> columns) throws SqlException {
```

Comment:
this should be done via `LowerCaseCharSequenceHashSet` to ensure:

- O(log n)
- case sensitivity is dealt with

Algorithm also needs to take into account column aliases where present

N.B. `LowerCaseCharSequenceHashSet` instance can be a class variable, which is reused and cleared after use

Reply:
Improved the complexity and added tests. 

Column aliases will further be discussed.